File: csharp/ShoutyReportProcessor.cs
using System.Collections.Generic;
using System.Xml;

class ShoutyReportProcessor
{
    IList<MileageClaim> mileageClaims;
    ShoutyStatsService statsService = new ShoutyStatsService();

    public ShoutyReportProcessor(IList<MileageClaim> mileageClaims)
    {
        this.mileageClaims = mileageClaims;
    }

    public void Process()
    {
       XmlDocument reportXml = new XmlDocument();
        reportXml.LoadXml("<?xml version=\"1.0\" encoding=\"UTF-8\"?><ecoReport/>");
        foreach (var claim in mileageClaims)
        {
            string responseXml = statsService.GetRevenueForCustomer(claim.CustomerID);

            var responseDocument = new XmlDocument();
            responseDocument.LoadXml(responseXml);
            var revenue = decimal.Parse(responseDocument.DocumentElement.Attributes["revenue"].Value);

            var revenuePerMile = (float)revenue / (float)claim.Miles;
            XmlElement node = reportXml.CreateElement(string.Empty, "ecoStat", string.Empty);
            node.SetAttribute("SalespersonName", claim.Name);
            node.SetAttribute("RevenuePerMile", System.Convert.ToString(revenuePerMile));
            reportXml.DocumentElement.AppendChild(node);
        }
 
        reportXml.Save("report.xml");
    }
}

File: java/ShoutyReportProcessor.java
import org.w3c.dom.Document;
import org.w3c.dom.Element;

import java.util.List;

class ShoutyReportProcessor {
    List<MileageClaim> mileageClaims;
    ShoutyStatsService statsService = new ShoutyStatsService();

    public ShoutyReportProcessor(List<MileageClaim> mileageClaims) {
        this.mileageClaims = mileageClaims;
    }

    public void process() {
        Document doc = XmlHelper.newDocument();
        Element ecoReport = doc.createElement("ecoReport");
        doc.appendChild(ecoReport);

        for (MileageClaim claim : mileageClaims) {
            String requestXml = "<Customer id=\"" + claim.customerID + "\"/>";
            String responseXml = statsService.getRevenueForCustomer(requestXml);
            Document responseDocument = XmlHelper.parse(responseXml);

            double revenue = Double.parseDouble(responseDocument.getDocumentElement().getAttribute("revenue"));
            float revenuePerMile = (float) revenue / (float) claim.miles;

            Element node = doc.createElement("ecoStat");
            node.setAttribute("SalespersonName", claim.name);
            node.setAttribute("RevenuePerMile", String.valueOf(revenuePerMile));
            ecoReport.appendChild(node);
        }


        XmlHelper.write(doc, "report.xml");
    }
}
